tests/backup/backup_helper.go

```diff
@@ -3259,19 +3259,30 @@ func IsMongoDBReady() error {
 		if err != nil {
 			return "", true, err
 		}
-		// Px-Backup would function with just 2 mongo DB pods in healthy state.
-		// Ideally we would expect all 3 pods to be ready but because of intermittent issues, we are limiting to 2
-		// TODO: Remove the limit to check for only 2 out of 3 pods once fixed
-		// Tracking JIRAs: https://portworx.atlassian.net/browse/PB-3105, https://portworx.atlassian.net/browse/PB-3481
-		if statefulSet.Status.ReadyReplicas < 2 {
+
+		// Check if all 3 mongo pods have come up
+		if statefulSet.Status.ReadyReplicas < 3 {
 			return "", true, fmt.Errorf("mongodb pods are not ready yet. expected ready pods - %d, actual ready pods - %d",
-				2, statefulSet.Status.ReadyReplicas)
+				3, statefulSet.Status.ReadyReplicas)
+
 		}
 		return "", false, nil
 	}
 	_, err = DoRetryWithTimeoutWithGinkgoRecover(mongoDBPodStatus, 30*time.Minute, 30*time.Second)
 	if err != nil {
-		return err
+		if strings.Contains(err.Error(), "mongodb pods are not ready yet") {
```

Comment:
Create a string variable for `mongodb pods are not ready yet`. In future if someone changes the string in line 3267 then we can avoid failure.

Reply:
done